index.js

```diff
@@ -12,6 +14,11 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = opts.fromIndex > 0 ? opts.fromIndex : 0;
+
+	if (startIndex > input.length) {
+		return input;
+	}
 
 	while (true) { // eslint-disable-line no-constant-condition
 		const index = input.indexOf(needle, prevIndex);
```

Comment:
After analyzing my last commit, I can only agree with you. On the first iteration, the variable `fromIndex` was not being used, thus defeating the purpose of this feature.